### PR TITLE
New repository documentation - contributing md file

### PR DESCRIPTION
**Pull request by zloyden**
https://github.com/AdguardTeam/AdguardFilters/issues/107129
Feel free to add something new or change current statements.

### Checklist

* [x] Add other our filters under the `Repository structure`.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Contributing to AdGuard filters
+
+Short introduction, explain what the ways to contribute are.
+
+## How to work with this repo
+
+1. Install [Visual Studio Code](https://code.visualstudio.com/download) with the ["Adblock" syntax plugin](https://marketplace.visualstudio.com/items?itemName=adguard.adblock).
+2. Clone (or fork and clone) [this repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository).
+3. Make your changes and commit them.
+4. Every commit needs to be linked to an issue. The commit message should look like (`Fix #123` where `123` is the issue number). This way the issue will be closed automatically once this commit lands in the `master` branch.
```
Comment (Alex-302) [inline CONTRIBUTING.md:10]: maybe better `Fix #123 site/comment`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## How to write filter rules
+
+Before you start, please also note we follow to the current [filters policy](https://kb.adguard.com/en/general/adguard-filter-policy). You should read and understand it - one of the most important things is the [quality requirements](https://kb.adguard.com/en/general/adguard-filter-policy#quality-requirements-for-filtering-rules).
```
Comment (Alex-302) [inline CONTRIBUTING.md:14]: Remove `/en/` from links. The site redirects to correct lang.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Contributing to AdGuard filters
+
+Short introduction, explain what the ways to contribute are.
+
+## How to work with this repo
+
+1. Install [Visual Studio Code](https://code.visualstudio.com/download) with the ["Adblock" syntax plugin](https://marketplace.visualstudio.com/items?itemName=adguard.adblock).
+2. Clone (or fork and clone) [this repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository).
```
Comment (AdamWr) [inline CONTRIBUTING.md:8]: Maybe it would be better to add a link to `Clone` word instead of `this repo`, because it may be a bit misleading, I think.

Comment (zloyden) [reply CONTRIBUTING.md]: Thank you, changed it )

File: CONTRIBUTING.md (hunk 1 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Ok, changed the point 4.

File: CONTRIBUTING.md (hunk 2 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Removed it.

**Comment by zloyden** [comment]
@Yuki2718 thanks for pointing this out. We have been discussing about the structure for now, after that the text will be reviewed,

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Contributing to AdGuard filters
+
+Short introduction, explain what the ways to contribute are.
```
[the remark this answers is not in the document]
Comment (anastasiia-fedotova) [reply CONTRIBUTING.md:3]: не new filters - a new rules

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+## How to work with this repo
+
+1. Install [Visual Studio Code](https://code.visualstudio.com/download) with the ["Adblock" syntax plugin](https://marketplace.visualstudio.com/items?itemName=adguard.adblock).
+2. [Clone](https://docs.github.com/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository) (or fork and clone) this repo.
+3. Make your changes and commit them.
+4. Every commit needs to be linked to an issue. The commit message should look like the example below. This way the issue will be closed automatically once this commit lands in the `master` branch.
+   * `Fix #123 site` where `123` is the issue number and `site` - website from the issue.
+   * `Fix #123 comment` where `comment` is your comment for additional fixes in the pull request.
```
[the remark this answers is not in the document]
Comment (anastasiia-fedotova) [reply CONTRIBUTING.md:12]: это не надо

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+## Repository structure
+
+AdGuard filters are combined from the files in this repository. This is an automated process that is done periodically by the scripts in the [FiltersRegistry](https://github.com/AdguardTeam/FiltersRegistry) repo. In this repository, every filter list is split into a number of files and every file has its own purpose. If you're adding a new rule, make sure it is added to the proper file.
+
+#### AdGuard Base filter
+
+* Purpose: blocking different type of ads mostly on English, multilingual sites.
+* [Base folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/EnglishFilter/sections)
+* Requirements for rule submissions: don't add the rules at the beginning (make some step near the beginning); if you add rules with task comment or hints, put them next to the same structure in the file.
```
[the remark this answers is not in the document]
Comment (anastasiia-fedotova) [reply CONTRIBUTING.md:30]: add the rules to the beginning of the file...

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+* There is an official [documentation](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) that can help you.
+* When you're done with creating rules, please take a look at the similar ones in the filters. This may help you to make a better version of the rule.
+
+## Repository structure
+
+AdGuard filters are combined from the files in this repository. This is an automated process that is done periodically by the scripts in the [FiltersRegistry](https://github.com/AdguardTeam/FiltersRegistry) repo. In this repository, every filter list is split into a number of files and every file has its own purpose. If you're adding a new rule, make sure it is added to the proper file.
+
+#### AdGuard Base filter
+
+* Purpose: blocking different type of ads mostly on English, multilingual sites.
```
[the remark this answers is not in the document]
Comment (zloyden) [reply CONTRIBUTING.md:28]: Да, относится к этому блоку.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+1. Install [Visual Studio Code](https://code.visualstudio.com/download) with the ["Adblock" syntax plugin](https://marketplace.visualstudio.com/items?itemName=adguard.adblock).
+2. [Clone](https://docs.github.com/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository) (or fork and clone) this repo.
+3. Make your changes and commit them.
+4. Each commit should be linked to an issue. The commit message should look like the examples below. That way, the issue will be closed automatically once the commit is merged into the `master` branch.
+   * `Fix #123 site` where `123` is the issue number and `site` - website from the issue.
+   * `Fix #123 comment` where `comment` is your comment for additional fixes in the pull request.
+
+## How to write filter rules
+
+Before you begin, please read and understand the current [filters policy](https://kb.adguard.com/general/adguard-filter-policy) we adhere to. One of its most important points is the [quality requirements](https://kb.adguard.com/general/adguard-filter-policy#quality-requirements-for-filtering-rules).
```
Comment (ameshkov) [inline CONTRIBUTING.md:16]: @zloyden should we update the quality requirements and maybe list them here?

As I understand, we should at least have different quality requirements for the rules that hide ad placeholders (there's no need to apply them on websites that are not very popular).

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+The next step is creating rules.
+   * There is an official [documentation](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) that can help you.
+   * When you're done with creating rules, please take a look at the similar ones in the filters. This may help you to make a better version of the rule.
+
+## Repository structure
+
+AdGuard filters are compiled from files in this repository. This is an automated process that is periodically run by scripts in the [FiltersRegistry](https://github.com/AdguardTeam/FiltersRegistry) repo.  In this repository, each filter list is divided into several files, and each file has its own purpose. If you're adding a new rule, make sure it is added to the proper file.
+
+#### AdGuard Base filter
```
Comment (ameshkov) [inline CONTRIBUTING.md:26]: @zloyden what about other filter lists?

**Review by ameshkov — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 8 above)
Comment (zloyden) [reply CONTRIBUTING.md:16]: > should we update the quality requirements and maybe list them here?

Maybe not, those statements are clear to me. @Alex-302 wanted to add some changes in general.

> there's no need to apply them on websites that are not very popular

Probably, I missed something. Can you please expain? We don't add any type of the rules if the website has low visitors, except cases with incorrect blocking or adblock detection.

File: CONTRIBUTING.md (hunk 9 above)
Comment (zloyden) [reply CONTRIBUTING.md:26]: Should we add them? If so, how much? I thought this is a short introduction and one example should be enough.

File: CONTRIBUTING.md (hunk 8 above)
Comment (ameshkov) [reply CONTRIBUTING.md:16]: There're some reports on ad placeholders left after blocking ads. These should be only accepted if the website is popular enough, it makes little sense to handle placeholders on websites with 30k monthly visitors.

File: CONTRIBUTING.md (hunk 9 above)
Comment (ameshkov) [reply CONTRIBUTING.md:26]: We definitely should list all our lists here.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# Contributing to AdGuard filters
+
+If you want to make AdGuard better by implementing new rules, follow the instructions below to make your ideas come to life faster!
```
Comment (vbagirov) [inline CONTRIBUTING.md:3]: Мне кажется "implement" подразумевает именно внедрение, а не сам факт создания правила. "Creating", "coming up with" - такие варианты кажутся лучше.

File: CONTRIBUTING.md (hunk 8 above)
Comment (zloyden) [reply CONTRIBUTING.md:16]: > it makes little sense to handle placeholders on websites with 30k monthly visitors
Ok, I got it. This one should be added in the filters policy. However, what should be the acceptable value of visitors in this case?

File: CONTRIBUTING.md (hunk 9 above)
Comment (zloyden) [reply CONTRIBUTING.md:26]: Ok, I'll add our filters.

File: CONTRIBUTING.md (hunk 8 above)
Comment (ameshkov) [reply CONTRIBUTING.md:16]: That's up to you all to decide

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+## How to work with the repo
+
+1. Install [Visual Studio Code](https://code.visualstudio.com/download) with the ["Adblock" syntax plugin](https://marketplace.visualstudio.com/items?itemName=adguard.adblock).
+2. [Clone](https://docs.github.com/repositories/creating-and-managing-repositories/cloning-a-repository#cloning-a-repository) (or fork and clone) this repo.
+3. Make your changes and commit them.
+4. Each commit should be linked to an issue. The commit message should look like the examples below. That way, the issue will be closed automatically once the commit is merged into the `master` branch.
+   * `Fix #123 site` where `123` is the issue number and `site` - website from the issue.
```
Comment (vbagirov) [inline CONTRIBUTING.md:11]: мне кажется лучше в примере написать типа `example.org`, а не просто `site`, а то будут писать типа "youtube" а не "youtube.com"

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+   * `Fix #123 comment` where `comment` is your comment for additional fixes in the pull request.
+
+## How to write filter rules
+
+Before you begin, please read and understand the current [filters policy](https://kb.adguard.com/general/adguard-filter-policy) we adhere to. One of its most important points is the [quality requirements](https://kb.adguard.com/general/adguard-filter-policy#quality-requirements-for-filtering-rules).
+
+The next step is creating rules.
+   * There is an official [documentation](https://kb.adguard.com/general/how-to-create-your-own-ad-filters) that can help you.
```
Comment (vbagirov) [inline CONTRIBUTING.md:19]: [official documentation] лучше

File: CONTRIBUTING.md (hunk 9 above)
Comment (vbagirov) [reply CONTRIBUTING.md:26]: Почему тут `####` а не `###`?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+## Repository structure
+
+AdGuard filters are compiled from files in this repository. This is an automated process that is periodically run by scripts in the [FiltersRegistry](https://github.com/AdguardTeam/FiltersRegistry) repo.  In this repository, each filter list is divided into several files, and each file has its own purpose. If you're adding a new rule, make sure it is added to the proper file.
+
+#### AdGuard Base filter
+
+* Purpose: this filter blocks various kinds of ads mostly on English-language and multilingual sites.
+* [Base folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/EnglishFilter/sections)
+* Requirements for submitting rules: don't add rules to the beginning of the file, start entering them from line 4, for example. If you add rules with a task comment or hints, put them next to the same structure in the file.
+* Notes: The AdGuard Base filter is supposed to be used with [Easylist](https://github.com/easylist/easylist) (like `AdGuard French` and `AdGuard Chinese`), so there's no need to add rules which are already in `Easylist`.
```
Comment (vbagirov) [inline CONTRIBUTING.md:31]: Именно supposed? Потому что в такой формулировке это выглядит как будто предполагается, что они в паре используются всегда.

**Review by vbagirov — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 10 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Поменял на `creating`.

File: CONTRIBUTING.md (hunk 11 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Добавил `example.org` вместо `site`.

File: CONTRIBUTING.md (hunk 13 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Написано так, поскольку зависит от используемого продукта. В ublock origin такого нет - там эти фильтры используются раздельно.
Уточнил предложение, в котором теперь акцент на использовании продуктов AdGuard.

**Comment by Yuki2718** [comment]
> The AdGuard Base filter includes Easylist (like `AdGuard French` and `AdGuard Chinese`) if you use AdGuard products, so there's no need to add rules which are already in `Easylist`.

The AdGuard Base filter includes Easylist (like `AdGuard French` does Liste FR and `AdGuard Chinese` EasyList China`) if you use AdGuard products, so there's no need to add rules which are already in these included lists.

**Comment by zloyden** [comment]
> Maybe better to list all the lists up which have such inclusion? e.g. AG German.

I've been adding all our available filters, so yes. I'll add a note to the filters like `AdGuard Base`.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -23,9 +23,87 @@ The next step is creating rules.
...
+* [Annoyances folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/AnnoyancesFilter/sections)
+
+### AdGuard Experimental filter
+
+* Purpose: this filter blocks various kinds of elements (ads, annoyances etc) on sites that need to be properly tested before moving the rules to other filters.
+* [Experimental folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/ExperimentalFilter/sections)
+
+### AdGuard Spyware filter
+
+* Purpose: this filter blocks various kinds of analytics, trackers on sites.
```
Comment (anastasiia-fedotova) [inline CONTRIBUTING.md:103]: различные виды аналитики? инструменты аналитики, инструменты сбора информации - типа того, может?

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -23,9 +23,87 @@ The next step is creating rules.
...
+* [Spyware folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/SpywareFilter/sections)
+
+### AdGuard TrackParam filter
+
+* Purpose: this filter removes various kinds of tracking parameters on sites.
```
Comment (anastasiia-fedotova) [inline CONTRIBUTING.md:108]: ...from sites

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -23,9 +23,87 @@ The next step is creating rules.
...
+* [Mobile folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/MobileFilter/sections)
+
+### AdGuard Social filter
+
+* Purpose: this filter blocks various kinds of social widgets on sites.
+* [Social folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/SocialFilter/sections)
+
+### AdGuard UsefulAdsFilter filter
+
+* Purpose: this filter shows useful ads on sites.
```
[the remark this answers is not in the document]
Comment (zloyden) [reply CONTRIBUTING.md]: Данный фильтр разблокирует рекламу, которая может быть полезна пользователям. Поменял описание.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -23,9 +23,87 @@ The next step is creating rules.
...
+* [UsefulAdsFilter folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/UsefulAdsFilter/sections)
+
+### AdGuard Annoyances filter
+
+* Purpose: this filter blocks various kinds of annoyances on sites.
+* [Annoyances folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/AnnoyancesFilter/sections)
+
+### AdGuard Experimental filter
+
+* Purpose: this filter blocks various kinds of elements (ads, annoyances etc) on sites that need to be properly tested before moving the rules to other filters.
```
[the remark this answers is not in the document]
Comment (zloyden) [reply CONTRIBUTING.md]: Поменял формулировку.

File: CONTRIBUTING.md (hunk 14 above)
Comment (zloyden) [reply CONTRIBUTING.md]: Поменял формулировку.

**Review by vbagirov — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+* [Annoyances folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/AnnoyancesFilter/sections)
+
+### AdGuard Experimental filter
+
+* Purpose: this filter serves to test some new filtering rules that can potentially cause conflicts and mess with websites' work. In case these rules perform without any issues, they get added to main filters.
+* [Experimental folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/ExperimentalFilter/sections)
+
+### AdGuard Spyware filter
+
+* Purpose: this filter hides your actions online and avoid tracking.
```
[the remark this answers is not in the document]
Comment (zloyden) [reply CONTRIBUTING.md]: I've changed the description of purpose.

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+* [Spyware folder](https://github.com/AdguardTeam/AdguardFilters/tree/master/SpywareFilter/sections)
+
+### AdGuard TrackParam filter
```
[the remark this answers is not in the document]
Comment (zloyden) [reply CONTRIBUTING.md]: Все правильно, я просто внутреннее название использовал. Поменял на `AdGuard URL Tracking`.

**Review by vbagirov — APPROVED** [review]

**Review by Alex-302 — APPROVED** [review]

**Pull request merged**